### PR TITLE
Fix #217

### LINKED ISSUE #217: Available releases should not list early access versions as most recent feature release
The available releases endpoint lists java 15 as most recent feature release:

```
INFORMATION: 1 * Sending client request on thread main
1 > GET https://api.adoptopenjdk.net/v3/info/available_releases

INFORMATION: 1 * Client response received on thread main
1 < 200
{
   ...
    "most_recent_feature_release": 15,
    "most_recent_lts": 11
}
```

When trying to access this feature release through the assets endpoint, no binaries are found:
```
INFORMATION: 2 * Sending client request on thread main
2 > GET https://api.adoptopenjdk.net/v3/assets/feature_releases/15/ga

INFORMATION: 2 * Client response received on thread main
2 < 404
```

As there are no general availability (ga), but only early access (ea) binaries for this feature release, the available releases endpoint should not list this version as `most_recent_feature_release`.

### PR DESCRIPTION
**Pull request by johnoliver**
# Checklist

- [ ] You added tests to cover the change
- [ ] `mvn clean install` build and test completes
- [ ] You changed or added to the documentation

### CODE REVIEW

File: adoptopenjdk-api-v3-frontend/src/test/kotlin/net/adoptopenjdk/api/AvailableReleasesPathTest.kt [hunk 1]
```diff
@@ -2,17 +2,46 @@ package net.adoptopenjdk.api
...
+            .then()
+            .body(object : TypeSafeMatcher<String>() {
+
+                override fun describeTo(description: Description?) {
+                    description!!.appendText("json")
+                }
+
+                override fun matchesSafely(p0: String?): Boolean {
+                    val releaseInfo = JsonMapper.mapper.readValue(p0, ReleaseInfo::class.java)
+                    return releaseInfo.most_recent_feature_release == 13
```
Comment (karianna) [inline adoptopenjdk-api-v3-frontend/src/test/kotlin/net/adoptopenjdk/api/AvailableReleasesPathTest.kt:43]: Is it 13?

Comment (johnoliver) [reply adoptopenjdk-api-v3-frontend/src/test/kotlin/net/adoptopenjdk/api/AvailableReleasesPathTest.kt:43]: it is in the test dataset

**Comment by johnoliver** [comment]
expanded this to also mostly address #212

**Comment by M-Davies** [comment]
Gets my thumbs up for the addition of `most_recent_feature_version`, thanks 👍

**Comment by karianna** [comment]
> expanded this to also mostly address #212 with the exception of tip_version

You want me to hold off landing this until you add tip_version or?

**Comment by johnoliver** [comment]
@karianna no, tip is actually a bit more complex question so I would do this now

**Pull request merged**